File: GameLibrary/src/cc/lib/swing/AWTPanel.java
package cc.lib.swing;

import java.awt.*;

import javax.swing.*;

public class AWTPanel extends JPanel {

    public AWTPanel(int rows, int cols, Component ... components) {
        this(new GridLayout(rows, cols), components);
    }

	public AWTPanel(LayoutManager lm, Component ... components) {
		super(lm);
		for (Component c : components) {
			add(c);
		}
	}

	public AWTPanel(Component ... components) {
		for (Component c : components) {
			add(c);
		}
	}
	
	public void addTop(Component comp) {
        add(comp, BorderLayout.NORTH);
    }
    public void addBottom(Component comp) {
        add(comp, BorderLayout.SOUTH);
    }
    public void addLeft(Component comp) {
        add(comp, BorderLayout.WEST);
    }
    public void addRight(Component comp) {
        add(comp, BorderLayout.EAST);
    }
}

File: SwingApps/src/cc/lib/swing/AWTPanel.kt
package cc.lib.swing

import java.awt.BorderLayout
import java.awt.Component
import java.awt.GridLayout
import java.awt.LayoutManager
import javax.swing.JPanel

open class AWTPanel : JPanel {
	constructor(rows: Int, cols: Int, vararg components: Component) : this(GridLayout(rows, cols), *components) {}
	constructor(lm: LayoutManager, vararg components: Component) : super(lm) {
		for (c in components) {
			add(c)
		}
	}

	constructor(vararg components: Component) {
		for (c in components) {
			add(c)
		}
	}

	fun addTop(comp: Component) {
		require(layout is BorderLayout)
		add(comp, BorderLayout.NORTH)
	}

	fun addBottom(comp: Component) {
		require(layout is BorderLayout)
		add(comp, BorderLayout.SOUTH)
	}

	fun addCenter(comp: Component) {
		require(layout is BorderLayout)
		add(comp, BorderLayout.CENTER)
	}

	fun addLeft(comp: Component) {
		require(layout is BorderLayout)
		add(comp, BorderLayout.WEST)
	}

	fun addRight(comp: Component) {
		require(layout is BorderLayout)
		add(comp, BorderLayout.EAST)
	}
}